tokio/tests/sync_watch.rs

```diff
@@ -368,3 +368,52 @@ async fn receiver_is_notified_when_last_sender_is_dropped() {
 
     assert!(t.is_woken());
 }
+
+#[tokio::test]
+async fn receiver_changed_is_cooperative() {
+    let (tx, mut rx) = watch::channel(());
+
+    drop(tx);
+
+    tokio::select! {
+        biased;
+        _ = async {
+            loop {
+                let _ = rx.changed().await;
+            }
+        } => {},
+        _ = tokio::task::yield_now() => {},
+    }
+}
+
+#[tokio::test]
+async fn receiver_wait_for_is_cooperative() {
+    let (tx, mut rx) = watch::channel(0);
+
+    drop(tx);
+
+    tokio::select! {
+        biased;
+        _ = async {
+            loop {
+                let _ = rx.wait_for(|val| *val == 1).await;
```

Comment:
```suggestion
                assert!(rx.wait_for(|val| *val == 1).await.is_err());
```

Reply:
Done in [d5a6b3b](https://github.com/tokio-rs/tokio/pull/6846/commits/d5a6b3b7ca753e26c760d9af74e3e206edf97056).